README.md

```diff
@@ -63,22 +63,124 @@ hello-world:
+  # All tagged commits on the `master` branch from `v0.1.0` (inclusive) onwards:
+  - git+https://github.com/aiidalab/aiidalab-hello-world.git@master:v0.1.0^..
+  # All tagged commits on the `master` branch from `v0.1.0` (exclusive) until `v1.0.0`:
+  - git+https://github.com/aiidalab/aiidalab-hello-world.git@master:v0.1.0..v1.0.0
+
+  # The following approaches are *not recommended*, but shown here for completeness:
+  # The latest commit on the repository's default branch
+  - git+https://github.com/aiidalab/aiidalab-hello-world.git
+  # The latest commit on the repository's develop branch:
+  - git+https://github.com/aiidalab/aiidalab-hello-world.git@develop
+```
+
+#### Environment specification
+
+The environment specification is typically directly parsed from the app repository, similar to how [repo2docker](https://repo2docker.readthedocs.io) parses repositories to derive instructions to create a docker image.
+This app store uses the repo2env tool to automatically parse the environment specification for a specific app release.
+
+In the current version, the following files are parsed:
+
+ - `setup.cfg` - The Python requirements listed in that file must be installed within the Python kernel executing the app.
+ - `requirements.txt` (only if setup.cfg does not specify requirements) - The Python requirements listed in that file must be installed within the Python kernel executing the app.
+
+The above listed files are parsed either from a hidden sub-directory called `.aiidalab/` or the repository root directory in that order.
+This means for the purpose of the AiiDAlab environment specification, a file `.aiidalab/requirements.txt` would be preferentially parsed over the `./requirements.txt`.
+If a `.aiidalab` directory exists, only files within that directory are parsed.
+
+We recommend that app developers test the specification on their app before a release.
+Simply install and then run `repo2env` on a local repository to see the environment specification:
+```
+repo2env /path/to/app/repo
+```
+See the [repo2env documentation](https://github.com/aiidalab/repo2env#readme) for details.
+
+To override the environment specification for one or more versions, simply add an explicit specification to the release.
+For example, to override the environment specification for version v1.0.0:
+```yaml
+hello-world:
+  releases:
+  # The tagged commits on the default branch will use the parsed environment specification:
+  - git+https://github.com/aiidalab/aiidalab-hello-world.git@:
+  # The environment specification for version v1.0.0 is overriden:
+  - url: git+https://github.com/aiidalab/aiidalab-hello-world.git@v1.0.0
+    environment:
+      python-requirements:
+        - aiidalab>=20.02.0b2
+        - some-missing-requirement
+```
+
+This is useful to retoractively override environment specifications if the in-repository specification via the requirements-files is incorrect or incomplete.
```

Comment:
I see why this would be useful. At the same time it means that the requirements of a specific version can change over time, which complicates things.
What happens if you `aiidalab install ...` a package that is already installed - does it check whether the dependencies have changed in the meanwhile?

In most other package managers, overriding the dependencies of an app would be reflected in a change in the version of the app (e.g. they might call it `<version>_2` or `<version>-1`).
Since this might be difficult here, one could also think of using a short hash of the corresponding list item here to get a unique version... that version would not need to be displayed to the user in most cases but I suspect it might be useful to have access to it down the line.

There is of course a slightly more philosophical point here, which is that here you're making changes to an app as the administrator of the app store rather than as the app developer.
Should app developers who want to fix broken versions of their app make PRs to the app registry or should they just release fixed versions of their app?
In any case, at this stage I think this is indeed more a philosophical point than of practical relevance (might change once the registry grows).